src/lib/y2network/sysconfig/interfaces_reader.rb

```diff
@@ -144,6 +146,20 @@ def add_interface(name, conn)
         interface_class = conn.virtual? ? VirtualInterface : FakeInterface
         @interfaces << interface_class.from_connection(name, conn)
       end
+
+      # Detects the renaming mechanism used by the interface
+      #
+      # @param name [String] Interface's name
+      # @return [Symbol,nil] :mac (MAC address), :bus_id (BUS ID) or nil (no renaming)
+      def renaming_mechanism_for(name)
+        rule = UdevRule.find_for(name)
+        return nil unless rule
+        if rule.parts.any? { |p| p.key == "ATTR{address}" }
+          :mac
+        elsif rule.parts.any? { |p| p.key == "KERNELS" }
+          :bus_id
+        end
```

Comment:
don't we kill some complex udev rules there? E.g. if customer has his nice handcrafted udev rule for network card, if it survive run of yast2-network?

Reply:
The current implementations kills them anyway.

Reply:
yep, but we should aim to behave better as I see many customers use fine tuned udev rules.